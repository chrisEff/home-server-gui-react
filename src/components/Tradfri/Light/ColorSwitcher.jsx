'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import {connect} from 'react-redux'
import {setTradfriBulbColor} from '../../../actions'

const ColorSwitcher = ({deviceId, bulbType, color, onChange}) => {
	if (bulbType === 'rgb') {
		return (
			<div className='color-switcher'>
				<div className={color === 'red'     ? 'color red active'     : 'color red'}     onClick={() => onChange(deviceId, 'red')}><div></div></div>
				<div className={color === 'green'   ? 'color green active'   : 'color green'}   onClick={() => onChange(deviceId, 'green')}><div></div></div>
				<div className={color === 'blue'    ? 'color blue active'    : 'color blue'}    onClick={() => onChange(deviceId, 'blue')}><div></div></div>
				<div className={color === 'yellow'  ? 'color yellow active'  : 'color yellow'}  onClick={() => onChange(deviceId, 'yellow')}><div></div></div>
				<div className={color === 'pink'    ? 'color pink active'    : 'color pink'}    onClick={() => onChange(deviceId, 'pink')}><div></div></div>
				<div className={color === 'purple'  ? 'color purple active'  : 'color purple'}  onClick={() => onChange(deviceId, 'purple')}><div></div></div>
				<div className={color === 'warm'    ? 'color warm active'    : 'color warm'}    onClick={() => onChange(deviceId, 'warm')}><div></div></div>
				<div className={color === 'neutral' ? 'color neutral active' : 'color neutral'} onClick={() => onChange(deviceId, 'neutral')}><div></div></div>
				<div className={color === 'cold'    ? 'color cold active'    : 'color cold'}    onClick={() => onChange(deviceId, 'cold')}><div></div></div>
			</div>
		)
	}
	if (bulbType === 'white-spectrum') {
		return (
			<div className='color-switcher spectrum'>
				<div className={color === 'warm'    ? 'color warm active'    : 'color warm'}    onClick={() => onChange(deviceId, 'warm')}><div></div></div>
				<div className={color === 'neutral' ? 'color neutral active' : 'color neutral'} onClick={() => onChange(deviceId, 'neutral')}><div></div></div>
				<div className={color === 'cold'    ? 'color cold active'    : 'color cold'}    onClick={() => onChange(deviceId, 'cold')}><div></div></div>
			</div>
		)
	}
	if (bulbType === 'white') {
		return (
			<div className='color-switcher white'>
				<div className='color neutral active'><div></div></div>
			</div>
		)
	}
	return <div/>
}

ColorSwitcher.propTypes = {
	deviceId: PropTypes.number.isRequired,
	bulbType: PropTypes.string.isRequired,
	color: PropTypes.string,
	onchange: PropTypes.func,
}

const mapStateToProps = (state, ownProps) => ({
})
const mapDispatchToProps = dispatch => ({
	onChange: (id, color) => dispatch(setTradfriBulbColor(id, color)),
})

export default connect(mapStateToProps, mapDispatchToProps)(ColorSwitcher)
export {ColorSwitcher}
