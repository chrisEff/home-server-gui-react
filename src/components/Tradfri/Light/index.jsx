'use strict'

const homeServerApi = require('../../../homeServerApi')

const React = require('react')
const PropTypes = require('prop-types')

const LightTitle = require('./LightTitle')
const BrightnessSlider = require('./BrightnessSlider')
const ColorSwitcher = require('./ColorSwitcher')
const LightSwitch = require('./LightSwitch')

class Light extends React.Component {

	static propTypes = {
		bulb: PropTypes.object.isRequired,
	}

	constructor (props) {
		super(props)
		this.state = {bulb: props.bulb}
	}

	render () {
		const bulb = this.state.bulb
		const className = bulb.state ? 'light on' : 'light'
		return (
			<div className={className + ' ' + bulb.color}>
				<LightTitle title={bulb.name} onChange={this.updateName} />

				<LightSwitch state={bulb.state} onchange={this.updateState}/>
				<BrightnessSlider brightness={bulb.brightness} onchange={this.updateBrightness}/>
				<ColorSwitcher bulbType={bulb.bulbType} color={bulb.color} onchange={this.updateColor}/>
			</div>
		)
	}

	updateName = async (name) => {
		const bulb = this.state.bulb
		await homeServerApi.put(`/tradfri/device/${bulb.id}/name/${name}`)

		bulb.name = name
		this.setState({bulb})
	}

	updateState = async (state) => {
		const bulb = this.state.bulb
		await homeServerApi.put(`/tradfri/device/${bulb.id}/state/${state}`)

		bulb.state = state
		this.setState({bulb})
	}

	updateBrightness = async (brightness) => {
		const bulb = this.state.bulb
		await homeServerApi.put(`/tradfri/device/${bulb.id}/brightness/${brightness}`)

		bulb.brightness = brightness
		this.setState({bulb})
	}

	updateColor = async (color) => {
		const bulb = this.state.bulb
		await homeServerApi.put(`/tradfri/device/${bulb.id}/color/${color}`)

		bulb.color = color
		this.setState({bulb})
	}
}

module.exports = Light
