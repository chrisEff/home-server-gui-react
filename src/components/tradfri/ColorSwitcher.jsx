'use strict'

const React = require('react')
const PropTypes = require('prop-types')

const ColorSwitcher = ({bulbType, color, onchange}) => {
	if (bulbType === 'rgb') {
		return (
			<div className='color-switcher'>
				<div className={color === 'red'     ? 'color red active'     : 'color red'}     onClick={() => onchange('red')}><div></div></div>
				<div className={color === 'green'   ? 'color green active'   : 'color green'}   onClick={() => onchange('green')}><div></div></div>
				<div className={color === 'blue'    ? 'color blue active'    : 'color blue'}    onClick={() => onchange('blue')}><div></div></div>
				<div className={color === 'yellow'  ? 'color yellow active'  : 'color yellow'}  onClick={() => onchange('yellow')}><div></div></div>
				<div className={color === 'pink'    ? 'color pink active'    : 'color pink'}    onClick={() => onchange('pink')}><div></div></div>
				<div className={color === 'purple'  ? 'color purple active'  : 'color purple'}  onClick={() => onchange('purple')}><div></div></div>
				<div className={color === 'warm'    ? 'color warm active'    : 'color warm'}    onClick={() => onchange('warm')}><div></div></div>
				<div className={color === 'neutral' ? 'color neutral active' : 'color neutral'} onClick={() => onchange('neutral')}><div></div></div>
				<div className={color === 'cold'    ? 'color cold active'    : 'color cold'}    onClick={() => onchange('cold')}><div></div></div>
			</div>
		)
	}
	if (bulbType === 'white-spectrum') {
		return (
			<div className='color-switcher spectrum'>
				<div className={color === 'warm'    ? 'color warm active'    : 'color warm'}    onClick={() => onchange('warm')}><div></div></div>
				<div className={color === 'neutral' ? 'color neutral active' : 'color neutral'} onClick={() => onchange('neutral')}><div></div></div>
				<div className={color === 'cold'    ? 'color cold active'    : 'color cold'}    onClick={() => onchange('cold')}><div></div></div>
			</div>
		)
	}
	if (bulbType === 'white') {
		return (
			<div className='color-switcher white'>
				<div className='color neutral active'><div></div></div>
			</div>
		)
	}
	return <div/>
}

ColorSwitcher.propTypes = {
	bulbType: PropTypes.string.isRequired,
	color: PropTypes.string,
	onchange: PropTypes.func,
}

module.exports = ColorSwitcher
