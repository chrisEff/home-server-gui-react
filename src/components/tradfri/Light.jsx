'use strict'

const api = require('../../../config').api
api.url = `${api.protocol}://${api.host}:${api.port}`

const React = require('react')
const PropTypes = require('prop-types')

const LightTitle = require('./LightTitle')
const BrightnessSlider = require('./BrightnessSlider')
const ColorSwitcher = require('./ColorSwitcher')
const LightSwitch = require('./LightSwitch')

class Light extends React.Component {

	static propTypes = {
		bulb: PropTypes.object.isRequired,
	}

	constructor (props) {
		super(props)
		this.state = {bulb: props.bulb}
	}

	render () {
		const bulb = this.state.bulb
		const className = bulb.state ? 'light on' : 'light'
		return (
			<div className={className + ' ' + bulb.color}>
				<LightTitle title={bulb.name} onChange={this.updateName} />

				<LightSwitch state={bulb.state} onchange={this.updateState}/>
				<BrightnessSlider brightness={bulb.brightness} onchange={this.updateBrightness}/>
				<ColorSwitcher bulbType={bulb.bulbType} color={bulb.color} onchange={this.updateColor}/>
			</div>
		)
	}

	updateName = async (name) => {
		const bulb = this.state.bulb
		await fetch(`${api.url}/tradfri/device/${bulb.id}/name/${name}?key=${api.key}`, {method: 'PUT'})

		bulb.name = name
		this.setState({bulb})
	}

	updateState = async (state) => {
		const bulb = this.state.bulb
		await fetch(`${api.url}/tradfri/device/${bulb.id}/state/${state}?key=${api.key}`, {method: 'PUT'})

		bulb.state = state
		this.setState({bulb})
	}

	updateBrightness = async (brightness) => {
		const bulb = this.state.bulb
		await fetch(`${api.url}/tradfri/device/${bulb.id}/brightness/${brightness}?key=${api.key}`, {method: 'PUT'})

		bulb.brightness = brightness
		this.setState({bulb})
	}

	updateColor = async (color) => {
		const bulb = this.state.bulb
		await fetch(`${api.url}/tradfri/device/${bulb.id}/color/${color}?key=${api.key}`, {method: 'PUT'})

		bulb.color = color
		this.setState({bulb})
	}
}

module.exports = Light
