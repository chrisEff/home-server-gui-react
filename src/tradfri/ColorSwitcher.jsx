module.exports = ({bulbType, color, onchange}) => {  
  if (bulbType === 'rgb') {
    return (
      <div className='color-switcher'>
        <div className={color === 'red'     ? 'color red active'     : 'color red'}     onClick={() => onchange('red')}/>
        <div className={color === 'green'   ? 'color green active'   : 'color green'}   onClick={() => onchange('green')}/>
        <div className={color === 'blue'    ? 'color blue active'    : 'color blue'}    onClick={() => onchange('blue')}/>
        <div className={color === 'yellow'  ? 'color yellow active'  : 'color yellow'}  onClick={() => onchange('yellow')}/>
        <div className={color === 'pink'    ? 'color pink active'    : 'color pink'}    onClick={() => onchange('pink')}/>
        <div className={color === 'purple'  ? 'color purple active'  : 'color purple'}  onClick={() => onchange('purple')}/>
        <div className={color === 'warm'    ? 'color warm active'    : 'color warm'}    onClick={() => onchange('warm')}/>
        <div className={color === 'neutral' ? 'color neutral active' : 'color neutral'} onClick={() => onchange('neutral')}/>
        <div className={color === 'cold'    ? 'color cold active'    : 'color cold'}    onClick={() => onchange('cold')}/>
      </div>
    )
  }
  if (bulbType === 'white-spectrum') {
    return (
      <div className='color-switcher spectrum'>
        <div className={color === 'warm'    ? 'color warm active'    : 'color warm'}    onClick={() => onchange('warm')}/>
        <div className={color === 'neutral' ? 'color neutral active' : 'color neutral'} onClick={() => onchange('neutral')}/>
        <div className={color === 'cold'    ? 'color cold active'    : 'color cold'}    onClick={() => onchange('cold')}/>
      </div>
    )
  }
  if (bulbType === 'white') {
    return (
      <div className='color-switcher white'>
        <div className='color neutral active' />
      </div>
    )
  }
  return <div/>
}
