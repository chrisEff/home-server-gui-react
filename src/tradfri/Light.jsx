'use strict'

const config = require('../../config')

const React = require('react')
const BrightnessSlider = require('./BrightnessSlider')
const ColorSwitcher = require('./ColorSwitcher')
const LightSwitch = require('./LightSwitch')

class Light extends React.Component {
	constructor (props) {
		super(props)
		this.state = {bulb: props.bulb}
	}

	render () {
		const bulb = this.state.bulb;
		const className = bulb.state ? 'light on' : 'light'
		return (
			<div className={className + ' ' + bulb.color}>
				<span className={bulb.name.length > 15 ? 'name long' : 'name'}>{bulb.name}</span>

				<LightSwitch state={bulb.state} onchange={this.updateState}/>
				<BrightnessSlider brightness={bulb.brightness} onchange={this.updateBrightness}/>
				<ColorSwitcher bulbType={bulb.bulbType} color={bulb.color} onchange={this.updateColor}/>
			</div>
		)
	}

	updateState = async (state) => {
		const bulb = this.state.bulb
		await fetch(
			`${config.api.protocol}://${config.api.host}:${config.api.port}/tradfri/device/${bulb.id}/state/${state}?key=${config.api.key}`,
			{method: 'PUT'}
		)

		bulb.state = state
		this.setState({bulb})
	}

	updateBrightness = async (brightness) => {
		const bulb = this.state.bulb
		await fetch(
			`${config.api.protocol}://${config.api.host}:${config.api.port}/tradfri/device/${bulb.id}/brightness/${brightness}?key=${config.api.key}`,
			{method: 'PUT'}
		)

		bulb.brightness = brightness
		this.setState({bulb})
	}

	updateColor = async (color) => {
		const bulb = this.state.bulb
		await fetch(
			`${config.api.protocol}://${config.api.host}:${config.api.port}/tradfri/device/${bulb.id}/color/${color}?key=${config.api.key}`,
			{method: 'PUT'}
		)

		bulb.color = color
		this.setState({bulb})
	}
}

module.exports = Light
