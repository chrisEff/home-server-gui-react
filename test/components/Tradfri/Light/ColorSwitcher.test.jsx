'use strict'

import React from 'react'
import {ColorSwitcher} from '../../../../src/components/Tradfri/Light/ColorSwitcher'

import {shallow} from 'enzyme'

describe.only('ColorSwitcher', () => {

	it.each([
		['red'],
		['green'],
		['blue'],
		['yellow'],
		['pink'],
		['purple'],
		['warm'],
		['neutral'],
		['cold'],
	])('should render the rgb switcher correctly when %s is active', (color) => {
		const wrapper = shallow(<ColorSwitcher deviceId={1} bulbType="rgb" color={color} />)
		expect(wrapper).toMatchSnapshot()
	})

	it.each([
		['warm'],
		['neutral'],
		['cold'],
	])('should render the white-spectrum switcher correctly when %s is active', (color) => {
		const wrapper = shallow(<ColorSwitcher deviceId={1} bulbType="white-spectrum" color={color} />)
		expect(wrapper).toMatchSnapshot()
	})

	it('should render the white switcher correctly', () => {
		const wrapper = shallow(<ColorSwitcher deviceId={1} bulbType="white" color="neutral" />)
		expect(wrapper).toMatchSnapshot()
	})

})
