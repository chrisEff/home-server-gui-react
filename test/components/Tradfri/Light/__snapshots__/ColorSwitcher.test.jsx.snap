// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSwitcher should render the rgb switcher correctly when blue is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-red"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-red_k554e1"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-green"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-green_22ph69"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-blue"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-blue_106q26s"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-yellow"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-yellow_1ywdg4u"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-pink"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-pink_ub5v9e"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-purple"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-purple_s6keop"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the rgb switcher correctly when cold is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-red"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-red_k554e1"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-green"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-green_22ph69"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-blue"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-blue_106q26s"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-yellow"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-yellow_1ywdg4u"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-pink"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-pink_ub5v9e"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-purple"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-purple_s6keop"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the rgb switcher correctly when green is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-red"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-red_k554e1"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-green"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-green_22ph69"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-blue"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-blue_106q26s"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-yellow"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-yellow_1ywdg4u"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-pink"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-pink_ub5v9e"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-purple"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-purple_s6keop"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the rgb switcher correctly when neutral is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-red"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-red_k554e1"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-green"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-green_22ph69"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-blue"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-blue_106q26s"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-yellow"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-yellow_1ywdg4u"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-pink"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-pink_ub5v9e"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-purple"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-purple_s6keop"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the rgb switcher correctly when pink is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-red"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-red_k554e1"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-green"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-green_22ph69"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-blue"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-blue_106q26s"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-yellow"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-yellow_1ywdg4u"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-pink"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-pink_ub5v9e"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-purple"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-purple_s6keop"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the rgb switcher correctly when purple is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-red"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-red_k554e1"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-green"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-green_22ph69"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-blue"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-blue_106q26s"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-yellow"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-yellow_1ywdg4u"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-pink"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-pink_ub5v9e"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-purple"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-purple_s6keop"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the rgb switcher correctly when red is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-red"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-red_k554e1"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-green"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-green_22ph69"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-blue"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-blue_106q26s"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-yellow"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-yellow_1ywdg4u"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-pink"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-pink_ub5v9e"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-purple"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-purple_s6keop"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the rgb switcher correctly when warm is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-red"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-red_k554e1"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-green"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-green_22ph69"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-blue"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-blue_106q26s"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-yellow"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-yellow_1ywdg4u"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-pink"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-pink_ub5v9e"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-purple"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-purple_s6keop"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the rgb switcher correctly when yellow is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-red"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-red_k554e1"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-green"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-green_22ph69"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-blue"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-blue_106q26s"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-yellow"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-yellow_1ywdg4u"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-pink"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-pink_ub5v9e"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-purple"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-purple_s6keop"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the white switcher correctly 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the white-spectrum switcher correctly when cold is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the white-spectrum switcher correctly when neutral is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;

exports[`ColorSwitcher should render the white-spectrum switcher correctly when warm is active 1`] = `
<div
  className="colorSwitcher_1osgznj"
>
  <div
    className="color_17zglbt"
    key="1-color-warm"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-warm_1wjcaqm"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-neutral"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-neutral_1hyp8n0"
    />
  </div>
  <div
    className="color_17zglbt"
    key="1-color-cold"
    onClick={[Function]}
  >
    <div
      className="inner_16vgkgb-o_O-cold_k70gul"
    />
  </div>
</div>
`;
